### ISSUE TITLE
Styling issue: If label classes exceed 3 then not all class labels can be seen

### ISSUE THREAD

**Issue by 2320sharon** [pyviz-topics/holodoodler#13]
# If the number of classes > 3, not all classes can be seen

![issue1_classes_label_small](https://user-images.githubusercontent.com/61564689/148139954-21711d69-cbf3-4064-8acb-459270ed977c.png)

When I was testing HoloDoodler with more than 3 classes I ran into an issue where I could not see all classes as shown above.  Fortunately, the fix was very easy, simply remove or comment out the line in Notebook Application that changes the label_class widget into RadioButtonGroup so that it remains a Selector widget. This allows all the classes to be selected mutually exclusive from each other.

![issue1_classes_label_small_fixed](https://user-images.githubusercontent.com/61564689/148140104-72911305-ade8-4832-83fb-23e3cd2c4af7.png)

Here is what the final result looks like.

![attempt2_more_classeselwha_2018-03-25-19-07-06_L8_rgb](https://user-images.githubusercontent.com/61564689/148140270-3bc5433a-0b2b-4d6d-be6e-c79c6994d886.png)

**Comment by maximlt** [comment]
Hi @2320sharon !

Yes you're right! The `RadioButtonGroup` isn't a very responsive widget and when more than a few classes are in use the buttons overflow. I was aware of that and decided to keep this widget type just for the sake of the presentation, keeping in mind that:
1. a `Select` widget could be used instead even if it isn't as nice to use (one more click)
2. Panel doesn't provide directly the widget that I would like to use, i.e. a set of buttons that behave like a toggle and that wrap nicely in their container, but provides the set of tools that allow to build such components. It's just a little bit more work.
3. The `RadioButtonGroup` provided by Panel actually comes from Bokeh where it has an [option to be oriented vertically](https://docs.bokeh.org/en/latest/docs/reference/models/widgets/groups.html#bokeh.models.RadioButtonGroup.orientation). It appears that this option isn't made available in Panel and I don't see any particular reason why this option couldn't be offered to Panel users.

Solution 3. could be a good compromise between solutions 1. and 3.

**Comment by 2320sharon** [comment]
Hi @maximlt !

Glad you were aware of this issue. I figured that the long-term goal was to allow the buttons to wrap around in the display so that the user could easily see all their label options. I think having the buttons oriented vertically and allowing the container to stretch to accommodate more labels would be great a solution.

**Comment by jbednar** [comment]
Yes, solution 3 sounds good in this case, along with filing a feature request on Bokeh for the RadioButtonGroup to be responsive. For speed and ease of use when doing a lot of segmentations, it's important for all the categories to be visible on screen so that you can simply click on the one you want rather than having them be hidden behind a selector. The RadioButtonGroup provides a nice way to do that when there are a small number of short labels, leaving all the other UI elements similarly easily accessible, but orienting them vertically would be a good second-best option when they don't all fit.

**Comment by maximlt** [comment]
I've added in this [commit](https://github.com/pyviz-topics/holodoodler/commit/c593711a9ea8df5587902bcb83655159fec00993) a widget that is more reactive and colorized.

![image](https://user-images.githubusercontent.com/35924738/148744642-6ca89f80-153f-498a-82e0-7f90cbc00349.png)

I've also submitted a [PR](https://github.com/holoviz/panel/pull/3083) to Panel that exposes the `orientation` parameter of the Bokeh Radio button group widget. If the solution I've provided with the custom widget doesn't prove to be robust enough (I've had some layout issues with the FlexBox layout) then I believe this second option could be a good and simple alternative.
